README.md

```diff
@@ -1,69 +1,107 @@
-- An open source wallet/dApp API controlled by the community
-- An open source StarkNet wallet discovery list moderated by the community
+- ❤️‍🩹 Allow Starknet dApps and wallets to seamlessly connect
+- 🪶 Lightweight and easy to use
+- 🏎 Fast integration, building and testing
+- ⚙️ Customizable and extensible
+- 🌍 Open source and controlled by the community
 
 ## Installation
+
 ```
 # using npm
-npm install get-starknet
+npm install get-starknet starknet@next
 
 # using yarn
-yarn add get-starknet
+yarn add get-starknet starknet@next
+
+# using pnpm
+pnpm add get-starknet starknet@next
 ```
 
 ## Usage for dApp developers
 
-If you were using getStarknet() before, simply replace the import line as below.
-
-```diff
--import { getStarknet } from "@argent/get-starknet"
-+import { getStarknet } from "get-starknet"
-```
-
-- Optional - customize the CSS of get-starknet to match your look&amp;feel
-- Optional - integrate with new API functions (e.g.: modify wallet list using custom sort/include/exclude, etc.)
+You can use the built-in UI to connect to any Starknet wallet as fast as
+possible like this:
 
-## Details
+```tsx
+import { connect, disconnect } from "get-starknet"
 
-### Flow
+return (
+  <button onClick={() => connect()}>Connect wallet</button>
+)
+```
 
-- Once a dApp wants to connect to a wallet, it calls `getStarknet()` (or optionally use the new `connect` API to control the order, included/excluded, etc.)
+### Advanced usage
+
+You can also choose to customize the UI by overwriting the CSS classes, or by
+implementing your very own UI. This is possible due to a split into a `core` and
+`ui` package. As a library author or dapp developer who wants to implement a
+custom UI, you can use the `core` package.
+
+```tsx
+import {
+  disconnect,
+  enable,
+  getAvailableWallets,
+  getDefaultWallet,
+  getDiscoveryWallets,
+  getLastConnectedWallet,
+  getPreAuthorizedWallets,
+} from "get-starknet-core"
+
+interface SnWalletResult {
+  // Returns all wallets available in the window object
+  getAvailableWallets: (
+    options?: GetWalletOptions,
+  ) => Promise<StarknetWindowObject[]>
+  // Returns only preauthorized wallets available in the window object
+  getPreAuthorizedWallets: (
+    options?: GetWalletOptions,
+  ) => Promise<StarknetWindowObject[]>
+  // Returns all wallets in existence (from discovery file)
+  getDiscoveryWallets: (options?: GetWalletOptions) => Promise<WalletProvider[]>
+  // Returns the last wallet connected when it's still connected
+  getLastConnectedWallet: () => Promise<StarknetWindowObject | null>
+  // Returns the default wallet
+  getDefaultWallet: () => Promise<StarknetWindowObject | null>
+  // Connects to a wallet
+  enable: (
+    wallet: StarknetWindowObject,
+    options?: {
+      starknetVersion?: "v3" | "v4"
+    },
+  ) => Promise<ConnectedStarknetWindowObject>
+  // Disconnects from a wallet
+  disconnect: (options?: { clearDefaultWallet?: boolean }) => Promise<void>
```

Comment:
what about `clearLastWallet?: boolean;`?

Reply:
`lastWallet` gets cleared on every explicit disconnect. I think that makes it way easier to manage the currently connected wallet and the default wallet.

Reply:
so i'm not sure what's the point of having a `lastWallet`, especially when there is no way to set `defaultWalletStore` explicitly.
1. dapp calls `connect`
2. user choose wallet_1
3. user wants to change a wallet
4. user clicks "disconnect" - dapp doesn't know that wallet_1 is now "default" (as it didn't mark it as one, nor the user) 
5. dapp calls `disconnect` without passing params
6. user clicks `connect` again -> dapp calls `connect` -> last wallet pops-up again, since wallet_1 is marked as "default".

I think there is an unclarity related to what's "last" wallet, what's "default" wallet (and how do dapps/users set/manage it), etc.

Reply:
previously - dapps called disconnect with "clearLastWallet" which made sense since the user explicitly asked to "drop" the last wallet and let him/her choose a wallet again, the "default" terminology is too ambiguous. 
originally I added it there as a placeholder for cross-dapps settings (TBD how/when)

Reply:
how it's now working:
`lastWallet` is used to keep track of the connected wallet. This allows dapps to use the same wallet without asking the user again and again when reloading.
`defaultWallet` is just used when in *never ask* mode. This way there is a greater probability that never ask works, while the default is to show the wallet picker in case `lastWallet` isn't defined 

Hope that's fine for you. We could rename `lastWallet -> connectedWallet` and `defaultWallet -> lastConnnected` but that would break backwards compatibility for users choice.

Reply:
correct me if i'm wrong, but silent mode (aka "never ask" mode) should be based on `isPreauthorized` state, e.g. connect the user automatically to first pre-authorized wallet - and in case there are multiple such wallets, sort them by last in-use -> shuffled).
`connectedWallet` == `lastConnected` == `lastWallet`, these are all the same from get-starknet POV since get-starknet doesn't manage the `enabled` state for wallets, it just matches ("connect") a dapp with a wallet **object**,
if the user picked a wallet (aka `lastWallet`) and decided not to approve its connection request, it's up to the dapp to decide whether to call `enable` later again (i.e. before invoking a tx) or to call `disconnect` and drop its "connection" with the last given wallet object upon a failed `enable` request.

(`defaultWallet` was only a placeholder for a  future cross-dapps solution (TBD how/when))

so I suggest to lean only on `lastWallet`; if it's there - it takes priority (within its "authorized" boundaries), and if the dapp calls "`disconnect`" explicitly, we shouldn't keep it (or at least give dapps control over whether to drop it or not) - because otherwise dapps will need to manage this state internally (which's pretty annoying)

Reply:
I think you missed the case of keeping a connection

imagine 2 preauthorized wallets on the same dapp, and you connect with braavos
you refresh the page and the dapp does a silent connect, it expects that braavos connects again.
Then you explicitly logout with the dapp and reload.
The dapp would try to silent connect (should not work, as you logged out explicitly), so we need to keep track of if a wallet is connected or not 


I agree that variable naming is terrible here, as I said I think we should rename `lastWallet -> connectedWallet` and `defaultWallet -> lastConnnected` but that would break backwards compatibility for users choice.

Reply:
```
Then you explicitly logout with the dapp and reload.
The dapp would try to silent connect (should not work, as you logged out explicitly), so we need to keep track of if a wallet is connected or not
```

if a dapp called logout **with** `clearLastConnected` (aka `clearDefaultWallet`) -> then attempt to silently connect again -> I would expect it to fail because there is no `lastWallet` wallet available anymore, and because we have 2 pre-authorized wallets, we can't assume/random which one will be chosen since it could lead to bad UX (i.e. my dapp-related assets are listed on Braavos, but my argent-x got chosen randomly this time)

if a dapp called logout **without** `clearLastConnected` (aka `clearDefaultWallet`) -> then attempt to silently connect again -> I would expect it to work and prioritize+pick-up the `lastWallet ` wallet.

so I don't see how `connectedWallet` is a `get-starknet` **state** other than a standard API for returning the currently-matched wallet object back to the dapp (so it won't be forced to save a reference by itself).
it seems to me that the only true `get-starknet` state is `lastWallet`, and anything else should be implemented on dapp end (i.e. one dapp would call logout but won't drop `lastConnected` state, another one will, depending on their own needs/flows/states).

Reply:
Okay, so I removed the connected logic and removed the default wallet state.
It's now just keeping track of lastConnectedWallet and the dapp can choose to forget about it on disconnect
The dapp would need to conditionally call silentconnect based on a self-managed state if a wallet should still be connected or not